### PR TITLE
style(Feed): update typings and propTypes usage

### PR DESCRIPTION
**Pull request by layershifter**
This PR is part of work for removing propTypes in production bundle (#524, #731).
Also, cleanups and updates typings for #1072.

### CODE REVIEW

File: src/views/Feed/index.d.ts [hunk 1]
```diff
@@ -128,18 +136,20 @@ interface FeedExtraProps {
   className?: string;
 
   /** Shorthand for primary content. */
-  content?: any;
+  content?: React.ReactNode;
 
   /** An event can contain additional information like a set of images. */
-  images?: any;
+  images?: boolean | Array<any>;
```
Comment (levithomason) [inline src/views/Feed/index.d.ts:142]: Should this be array of React.ReactNode?

**Review by levithomason — COMMENTED** [review]
Just one comment regarding the image shorthand array type.

File: src/views/Feed/index.d.ts (hunk 1 above)
Comment (layershifter) [reply src/views/Feed/index.d.ts:142]: We set `React.ReactNode` when we deal with `contentShorthand`, but in this case we have:
```js
PropTypes.oneOfType([
  PropTypes.bool,
  customPropTypes.collectionShorthand,
]),
```

Which represents array of `itemShorthand`, we mark it as `any` for now.

I want to add new TS types for `customPropTypes` in future.

Comment (layershifter) [reply src/views/Feed/index.d.ts:142]: #1289

**Comment by levithomason** [comment]
Looks great, we'll follow up on types for customPropTypes in #1289 .

**Comment by levithomason** [comment]
Released in `semantic-ui-react@0.64.8`.

**Pull request merged**